x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -130,17 +131,7 @@ synchronized void updateWithFailure(Exception exc) {
         //note that when search fails, we may have gotten partial results before the failure. In that case async
         // search will return an error plus the last partial results that were collected.
         this.isPartial = true;
-        ElasticsearchException[] rootCauses = ElasticsearchException.guessRootCauses(exc);
-        if (rootCauses == null || rootCauses.length == 0) {
-            this.failure = new ElasticsearchException(exc.getMessage(), exc) {
-                @Override
-                protected String getExceptionName() {
-                    return getExceptionName(getCause());
-                }
-            };
-        } else {
-            this.failure = rootCauses[0];
-        }
+        this.failure = ExceptionsHelper.convertToElastic(exc);
```

Comment:
I think this is simpler and even preserves status codes, not sure why we were using guessRootCauses

Reply:
++